pkg/azuredevops/smoke/smoketests.go

```diff
@@ -0,0 +1,401 @@
+			if strings.Contains(line, expectedMissingMessage) {
+				return false, fmt.Errorf("unexpected message found in logs: %s", expectedMissingMessage)
+			}
+		}
+	}
+
+	return true, nil
+}
+
+func RunBuildTest(ctx context.Context, connection *azuredevops.Connection, projectName, pipelineName string, pipelineID int, test BuildTest) bool {
+	var pass bool
+	var err error
+
+	if test.ExpectAbsent {
+		pass, err = checkSpecificBuildForMissingCommand(ctx, connection, projectName, pipelineName, test.LogMessage, pipelineID)
+	} else {
+		pass, err = checkSpecificBuildForCommand(ctx, connection, projectName, pipelineName, test.LogMessage, pipelineID)
+	}
+
+	if err != nil {
+		log.Fatalf("Test failed for %s: %v\n", test.Description, err)
+	}
+
+	if !pass {
+		log.Fatalf("Test failed for %s: condition not met\n", test.Description)
+	}
+
+	fmt.Printf("Test passed for %s\n", test.Description)
+	return true
+}
+
+func RunTimelineTests(ctx context.Context, connection *azuredevops.Connection, projectName string, buildID int, test TimelineTest) bool {
+	var pass bool
+	var err error
+
+	pass, err = getBuildStageStatus(ctx, connection, projectName, buildID, test)
+	if err != nil {
+		log.Fatalf("Test failed for %s: %v\n", test.Name, err)
+	}
+
+	if !pass {
+		log.Fatalf("Test failed for %s: condition not met\n", test.Name)
+	}
+
+	fmt.Printf("Test passed for %s\n", test.Name)
+	return true
+}
+
+func ShouldRunTest(testsToRun string, testsToRunList []string, testName string) bool {
+	if testsToRun == "all" || testsToRun == "" {
+		return true
+	}
+	return slices.Contains(testsToRunList, testName)
+}
+
+func GetBuildTests() []BuildTest {
+	return []BuildTest{
+		{
+			Description:  "Checkout self repository",
+			LogMessage:   "Repository 'self' has been successfully checked out.",
+			ExpectAbsent: false,
+		},
+		{
+			Description:  "Checkout kyma-modules repository",
+			LogMessage:   "Repository 'kyma-modules' has been successfully checked out.",
+			ExpectAbsent: false,
+		},
+		{
+			Description:  "Checkout security-scans-modular repository",
+			LogMessage:   "Repository 'security-scans-modular' has been successfully checked out.",
+			ExpectAbsent: false,
+		},
+		{
+			Description:  "Verify the download of conduit-cli",
+			LogMessage:   "conduit-cli has been successfully downloaded and is executable.",
+			ExpectAbsent: false,
+		},
+		{
+			Description:  "Verify Python Installation",
+			LogMessage:   "command not found: python3",
+			ExpectAbsent: true,
+		},
+		{
+			Description:  "Verify gcloud Installation",
+			LogMessage:   "command not found: gcloud",
+			ExpectAbsent: true,
+		},
+		{
+			Description:  "Verify Go Installation",
+			LogMessage:   "command not found: go",
+			ExpectAbsent: true,
+		},
+		{
+			Description:  "Download Kyma cli",
+			LogMessage:   "The 'kyma' binary has been successfully downloaded and is executable.",
+			ExpectAbsent: false,
+		},
+	}
+}
+
+func GetTimelineTests() []TimelineTest {
```

Comment:
TimelineTests definitions should be maintained in separate file as an yaml or json. This way we can maintain definitions in separate from a tool, even have it in more than one location if that would make sense for us.